src/groupeddatatable/grouping.jl

```diff
@@ -236,14 +236,29 @@ colwise(sum, groupby(dt, :a))
 ```
 
 """
-colwise(f::Function, d::AbstractDataTable) = Any[vcat(f(d[idx])) for idx in 1:size(d, 2)]
+function colwise(f::Function, d::AbstractDataTable)
+    x = [f(d[i]) for i in 1:ncol(d)]
+    if eltype(x) <: Nullable
+        return NullableArray(x)
+    else
+        return x
+    end
+end
 colwise(f::Function, gd::GroupedDataTable) = map(colwise(f), gd)
 colwise(f::Function) = x -> colwise(f, x)
 colwise(f) = x -> colwise(f, x)
 # apply several functions to each column in a DataTable
-colwise{T<:Function}(fns::Vector{T}, d::AbstractDataTable) =
-    reshape(Any[vcat(f(d[idx])) for f in fns, idx in 1:size(d, 2)],
-            length(fns)*size(d, 2))
+function colwise{T<:Function}(fns::Vector{T}, d::AbstractDataTable)
+    res = Array{AbstractVector}(ncol(d))
+    for i in eachindex(res)
```

Comment:
You should be able to `@inbounds` this loop. Also, since we know that `res` is a standard array and thus indexing won't throw us any curveballs, using `1:ncol(d)` may be very marginally more efficient, so that `eachindex` doesn't have to compute the valid indices. (I imagine the difference would be marginal at best, but hey.) If you do that you should store `n = ncol(d)` (or whatever) ahead of time to avoid recomputing it.